### ISSUE TITLE
Do we need a separate crowdsec agent running?

### ISSUE THREAD

**Issue by apxcarter** [hslatman/caddy-crowdsec-bouncer#40]
I'm new to crowdsec, but my understanding is that I need an agent running on the same machine/vm/container to consume logs and send alerts to the LAPI. Does this also handle that for Caddy logs or is it just the bouncer?

**Comment by hslatman** [comment]
Hey @apxcarter,

This is just the Caddy bouncer, currently. I started looking at integrating the new App Sec component here, but I haven't had much time to think it all through, so that's not there yet.

You're right that you'll need the CrowdSec Agent running somewhere too; it's not launched as part of this Caddy bouncer, or something like that. There's some official docs for that part on the hub: https://app.crowdsec.net/hub/author/crowdsecurity/configurations/caddy-logs.  It shouldn't be required to run it on the same machine as Caddy, as long as the logs can be read by the Agent. You could send the logs to the Agent [using the syslog format](https://docs.crowdsec.net/docs/data_sources/syslog/), for example, although that probably requires some more configuration and testing to get right. Running it locally, or at least close (e.g. in a Docker image in the same network, potentially with a shared volume for the logs), probably is the easiest way to get started testing.

**Comment by Simbiat** [comment]
Don't know if it will be of much use, but my project https://github.com/Simbiat/simbiat.ru has `docker-compose.yml`, `env.example` and `config` folder with what is required to setup FrankenPHP, CrowdSec and MariaDB. No a tutorial, yes, but should be mostly straightforward.

**Comment by modem-man-gmx** [comment]
> Don't know if it will be of much use, but my project https://github.com/Simbiat/simbiat.ru has `docker-compose.yml`, `env.example` and `config` folder ...

After reading your statement, I was very interested in reading your project. But your README.md is very, very short. Not really attractive this time. Perhaps later.
дружба,
M.

**Comment by Simbiat** [comment]
@modem-man-gmx do you really need in a readme for code of a website, that can be viewed live 😅

**Comment by modem-man-gmx** [comment]
> [@modem-man-gmx](https://github.com/modem-man-gmx) do you really need in a readme for code of a website, that can be viewed live 😅

Yes, of course! You are surely able to read your own code and see the ideas behind. I'm able to read *my* own code and remember the ideas behind, but why should it work crosswise? ;-)

I am out of web server business when inline-PHP scripts started replacing static Pearl-generated sites. So aside of some old (and dangerous) knowledge, I am a beginner with modern security concepts and the whole Eco-system around. My expert knowledge is far away of modern web servers, of PHP and so on. Last time I worked with web servers, I wrote an own implementation in C99 on a µC. Guess, even Apache was not well known that time.

Nowadays I have to face the fact to been working decades with other technologies and need to learn and understand Caddy and Crowdsec (or alike) now quickly. That's why I am lingering around an asking questions.
